tasking/pulp/tasking/base.py

```diff
@@ -388,7 +388,7 @@ def cancel(task_id):
         # If the task is already done, just stop
         msg = _('Task [%(task_id)s] already in a completed state: %(state)s')
         _logger.info(msg % {'task_id': task_id, 'state': task_status.state})
-        return
+        raise PulpException(msg % {'task_id': task_id, 'state': task_status.state})
```

Comment:
This may warrant some discussion. If a task has already been canceled, and a user requests again for it to be canceled, is there any case where they will care that it was already canceled? Is that an error, or an exceptional circumstance?